src/test/java/org/rulez/demokracia/pdengine/VoteTest.java

```diff
@@ -29,9 +29,22 @@ public void cast_vote_deletes_our_ballot() {
 		String ballot = voteManager.obtainBallot(adminInfo.voteId, adminInfo.adminKey);
 		List<RankedChoice> theCastVote = new ArrayList<RankedChoice>();
 		Vote vote = getTheVote();
+		vote.canVote = true;
 		voteManager.castVote(adminInfo.voteId,ballot,theCastVote);
-		assertFalse(vote.ballots.contains(ballot));
-		
+		assertFalse(vote.ballots.contains(ballot));		
 	}
 
+	
+	@tested_feature("Vote")
+	@tested_operation("Cast vote")
+	@tested_behaviour("works only if canVote is true")
+	@Test
+	public void cast_vote_does_not_deletes_our_ballot() {
```

Comment:
cast vote does delete our ballot, if it can work.
so maybe something like "cast vote does not delete ballot if canVote is false" would be better.
There are also other behaviours to test in the canVote==false case:
- a ReportedError exception is thrown
- with an appropriate descriptive message
- the cast vote does not get recorded

Reply:
@magwas : Ok, so I will rename the tests to cast_vote_deletes_ballot_if_canVote_is_true and cast_vote_does_not_delete_ballot_if_canVote_is_false.
I have to add a ReportedError exception with "Cast vote does not get recorded" message, right?

Reply:
Yes, yes.